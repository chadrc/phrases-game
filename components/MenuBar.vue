<template>

    <nav class="navbar" role="navigation" aria-label="main navigation">
      <div class="navbar-brand">
        <span class="navbar-item">
          <h2 class="title">Phrases</h2>
        </span>

        <a role="button" class="navbar-burger burger" aria-label="menu" aria-expanded="false" data-target="navbarBasicExample">
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>

      <div id="navbarBasicExample" class="navbar-menu">
        <div class="navbar-start">
          <nuxt-link to="/" class="navbar-item">
            Home
          </nuxt-link>
          <nuxt-link to="/play" class="navbar-item">
            Play
          </nuxt-link>
          <nuxt-link to="/create" class="navbar-item">
            Create
          </nuxt-link>
        </div>

        <div class="navbar-end">
          <div class="navbar-item">
            <div class="buttons">
              <button class="button is-primary" @click="startSignUp">
                <strong>Sign up</strong>
              </button>
              <a class="button is-light">
                Log in
              </a>
            </div>
          </div>
        </div>
      </div>
      <AuthModal :active="signingUp" title="Sign Up" submit-text="Sign Up" @cancel="endSignUp" @submit="submitSignUp"/>
    </nav>
</template>

<script>
import AuthModal from "@/components/AuthModal";
import { mapActions, mapGetters } from "vuex";

export default {
  components: { AuthModal },
  computed: {
    ...mapGetters("auth", ["signingUp"])
  },
  methods: {
    ...mapActions("auth", ["startSignUp", "endSignUp", "submitSignUp"])
  },
};
</script>

<style scoped>

</style>
