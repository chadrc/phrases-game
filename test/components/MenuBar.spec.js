import MenuBar from "@/components/MenuBar.vue";
import { makeVueMock } from "@/test/utils";

describe("MenuBar", () => {
  const { wrapper } = makeVueMock(MenuBar, {
    shallow: true
  });

  test("is vue instance", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders regular", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
