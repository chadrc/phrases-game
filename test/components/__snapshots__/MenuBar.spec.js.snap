// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar has user 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
            style="display: none;"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <a
            class="button is-light"
            style="display: none;"
          >
            
            Log in
          
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    canceltext="Cancel"
    errormessage=""
    submittext="Sign Up"
    title="Sign Up"
  />
</nav>
`;

exports[`MenuBar renders regular 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  <div
    class="navbar-brand"
  >
    <span
      class="navbar-item"
    >
      <h2
        class="title"
      >
        Phrases
      </h2>
    </span>
     
    <a
      aria-expanded="false"
      aria-label="menu"
      class="navbar-burger burger"
      data-target="navbarBasicExample"
      role="button"
    >
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
       
      <span
        aria-hidden="true"
      />
    </a>
  </div>
   
  <div
    class="navbar-menu"
    id="navbarBasicExample"
  >
    <div
      class="navbar-start"
    >
      <nuxt-link-stub
        class="navbar-item"
        to="/"
      >
        
        Home
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/play"
      >
        
        Play
      
      </nuxt-link-stub>
       
      <nuxt-link-stub
        class="navbar-item"
        to="/create"
      >
        
        Create
      
      </nuxt-link-stub>
    </div>
     
    <div
      class="navbar-end"
    >
      <div
        class="navbar-item"
      >
        <div
          class="buttons"
        >
          <button
            class="button is-primary"
          >
            <strong>
              Sign up
            </strong>
          </button>
           
          <a
            class="button is-light"
          >
            
            Log in
          
          </a>
        </div>
      </div>
    </div>
  </div>
   
  <authmodal-stub
    canceltext="Cancel"
    errormessage=""
    submittext="Sign Up"
    title="Sign Up"
  />
</nav>
`;
